Code/Framework/AzCore/AzCore/Serialization/Json/JsonUtils.h

```diff
@@ -79,12 +79,28 @@ namespace AZ
         AZ::Outcome<rapidjson::Document, AZStd::string> ReadJsonStream(IO::GenericStream& stream);
         
         //! Load object with known class type
+        //! Even if errorsOut contains errors, the load to an object could have succeeded
+        AZ::Outcome<void, AZStd::string> LoadObjectFromStringByType(void* objectToLoad, const Uuid& objectType, AZStd::string_view source,
```

Comment:
if it is a valid use case, should we introduce a data structure instead of a string parameter, so it can be easily extended in the future.

Reply:
That's a perfectly valid question we should throw at future development of the JSON serialization system. The intention is to have it load with lots of forgiveness for old, bad, missing, extra data. But perhaps after the load, the could be an easily scraped structure for callers to iterate over and make changes/corrections to their loaded data. There are couple of mechanism to do some post-processing work with custom serializers based on type, and listeners for specific types. But the idea behind the switch from object stream (and I agree with it) is to just read the data and construct the object (or read object write data) without too much back and forth between callbacks and systems interfering with that flow.